alerts/cluster-monitoring-operator/PrometheusRemoteStorageFailures.md

```diff
@@ -0,0 +1,47 @@
+# PrometheusRemoteStorageFailures
+
+## Meaning
+
+The `PrometheusRemoteStorageFailures` alert triggers when there has been
+a constant increase in failures sending samples to the remote storage for more
+than 15 minutes, in either the platform Prometheus or a user-defined Prometheus
+instance.
+
+## Impact
+
+Prometheus samples in remote storage might be missing or inaccurate.
+
+## Diagnosis
+
+1. Determine if the alert has been triggered for the instance of Prometheus
+   instance used for default cluster monitoring or for the instance that
+   monitors user-defined projects by checking the alert message's `namespace`
+   label. The namespace value will indicate the Prometheus instance.
+
+   E.g. `openshift-monitoring` for default platform monitoring and
+   `openshift-user-workload-monitoring` for user-defined monitoring.
+
+2. Review the logs for the affected Prometheus instance:
+
+   ```console
+   $ NAMESPACE='<value of namespace label from alert>'
+
+   $ oc -n $NAMESPACE logs -l 'app.kubernetes.io/component=prometheus'
+   level=error ... msg="Failed to send batch, retrying" ...
+   ```
+
+3. Check prometheus logs and remote storage logs.
+
+4. Investigate network issues. Check configs and credentials.
+
+## Mitigation
```

Comment:
I would classify possible errors like:
* Failure to authenticate to the remote storage system => check the authentication parameters in the CMO and user-workload configmaps.
* Requests hitting rate-limits => tweak the queue configuration in the CMO and user-workload configmaps and/or limit the number of samples being sent.
* 5xx HTTP errors => check the remote storage system's logs.

Reply:
added this as some sample debugging guidelines.

Reply:
I'd love to see @simonpasquier 's examples more explicitly. Maybe something along the lines of

```
This alert indicates an issue in Prometheus' communication with the remote system.
The cause can be on either the Prometheus side or the remote side. Common issues are:
- _what Simon said_
```

Reply:
I put that above the Diagnosis, but I'll add that specific comment there as well, thanks!